src/EditorFeatures/Core/Implementation/Peek/PeekableItemSource.cs

```diff
@@ -65,13 +65,21 @@ public void AugmentPeekSession(IPeekSession session, IList<IPeekableItem> peekab
 
                 if (!document.SupportsSemanticModel)
                 {
-                    // For documents without semantic models, just try to use the goto-def service
-                    // as a reasonable place to peek at.
-                    var goToDefinitionService = document.GetLanguageService<IGoToDefinitionService>();
-                    var navigableItems = goToDefinitionService.FindDefinitionsAsync(document, triggerPoint.Value.Position, cancellationToken)
-                                                              .WaitAndGetResult(cancellationToken);
+                    var peekDefinitionService = document.GetLanguageService<IPeekDefinitionService>();
+                    if (peekDefinitionService != null)
+                    {
+                        results = peekDefinitionService.GetDefinitionPeekItems(document, triggerPoint.Value.Position, cancellationToken);
+                    }
+                    else
+                    {
+                        // For documents without semantic models, just try to use the goto-def service
```

Comment:
this comment doesn't seem accurate (it talks about semantic models... but that doesn't seem to apply here)...

(note: i recognize that comment existed before... but it still seems wrong :)).

Reply:
I think it was using that as shorthand for languages that aren't either C# or VB. :)

Reply:
Also, it checks `SupportsSemanticModel` immediately above, presumably because it requests one immediately in the `else` case.